### ISSUE TITLE
edit data in the style of the navicat

### ISSUE THREAD

**Issue by soycamiloypunto** [Alecaddd/sequeler#260]
The app is excellent, its possible edit data in the style of the navicat directly in table?

**Comment by paulodiovani** [comment]
Never used navicat, but if youre talking about in-place-editing, I would love that too.
Just please make issue summary and description more clear.

**Comment by wout** [comment]
The people behind [TablePlus](https://tableplus.io/) are [working on a Linux version](https://github.com/TablePlus/TablePlus/issues/1056). When they release it, I think TablePlus will become the default database tool for Linux.

Don't get me wrong, I love Sequeler, but it's a crippled experience not being able to edit or filter data.

**Comment by Alecaddd** [comment]
@wout uh, that looks nice, I'm really curious to try it.
It's fine by me, and actually it's better so I can see how they approach and solve the problems I'm having.

I didn't built Sequeler to be a commercial tool, but rather to be a useful and native GTK app for what I need.
Since I can work on this during spare weekends, progress and improvements are slow, but sooner or later all the missing features will be added.

Maybe having an awesome competitor will push me to do more :smile:

**Comment by wout** [comment]
@Alecaddd I've been following the progress on Sequeler and I am impressed. :smile:

**Comment by Alecaddd** [comment]
I'm working on that actually, but I'm stuck with two major problems:
1. The Gtk.TreeView is not that flexible. I can make those cells editable but the usability is awful and more than often data gets lost.
2. The limitations of LibGDA don't allow me to properly handle dates, JSON, timestamps, etc.

To solve this problems, I started working on a custom GTK data grid, which will allow me to handle cells and data as standalone elements and add all the features I want, without any limitations.
https://github.com/Alecaddd/sequeler/tree/new-treeview

The problem is that this solution is really heavy and slow compared to the native TreeView.
But anyway, I'll keep working on it :smile:

**Comment by wout** [comment]
That's great to hear! I really appreciate your work on this app. And the work done by the elementaryOS community as a whole. I am researching Vala in my spare time, so maybe I will be able to contribute myself when time allows. :wink:

**Comment by Alecaddd** [comment]
If you don't know it already, I'm publishing some Vala tutorials on YouTube: https://www.youtube.com/playlist?list=PLriKzYyLb28mn2lS3c5yqMHgLREi7kR9-

**Comment by paulodiovani** [comment]
@Alecaddd 

> I can make those cells editable but the usability is awful

I don't think _in place editting_ is mandatory.
For example, double click on a row could open an overlay window with the record to edit, focusing on the clicked cell.

**Comment by wout** [comment]
I agree with @paulodiovani. While _in place editing_ is nice, more than often it results in bad UX. Especially if table content is larger than the cell, which it is quite often actually. Except for numbers, I would prefer an overlay window. It would make things a lot easier with a datetime field for example. An overlay with a date and time selector would be much more appealing than being able to edit in place, manually.

In fact, inline editing is one of the downsides of TablePlus. :smiley:

**Comment by huyphams** [comment]
@wout you can right-click => quick look, and it allows you to edit with a big popup Window or you can also edit it in the right detail pane.

**Comment by Zebouski** [comment]
Would absolutely love support for this in sequeler. Sequeler has been performant and uses a native linux GUI.

I've moved away to beekeeper for now which isn't performant cause electron... but I'll come right back if sequeler gets this.

Are there any developers left on this project? Could we put together a bountysource for some of these critical features to make this the ultimate casual SQL client for linux?
